### DISCUSSION TITLE
How to use: inverter.fit_sandia

### DISCUSSION THREAD

**Discussion by tonival94** [pvlib/pvlib-python#2327] [Q&A] [answered]
hi guys,

I am trying to understand how to use the function inverter.fit_sandia.
I want to use the function to get the sandia parameter to fit the 3 efficiency curves provided in the OND.

how should my data be formatted to be working with the function?

**Comment by cwhanse** [comment, marked as answer]
The docstring summarizes the data expected by that function:

```
    Parameters
    ----------
    ac_power : array_like
        AC power output at each data point [W].
    dc_power : array_like
        DC power input at each data point [W].
    dc_voltage : array_like
        DC input voltage at each data point [V].
    dc_voltage_level : array_like
        DC input voltage level at each data point. Values must be 'Vmin',
        'Vnom' or 'Vmax'.
```

"array_like" can be a numpy array or a pandas Series. Lists won't work.

Each array needs to be the same length. The values in each array must be ordered so that the values for one measurement point are in the same position in each array. The easiest way to do this is probably to put your data into a DataFrame, and then pass a Series to each corresponding inputs.

**Reply by tonival94** [reply]
But how about the dc_voltage_level variable?

I should have a 'low', 'mid', 'high' numpy array with same length as other input variable?

**Reply by cwhanse** [reply]
The inputs expect the arrays to be "stacked" so that the `dc_voltage` vector includes all the voltages (low, mid and high). The `dc_voltage_level` labels each position in the array so that the fitting can separate out data for each voltage level. We used that structure because it was consistent with some data we had from a lab's inverter test system.

**Reply by tonival94** [reply]
Thanks mate!
